pkg/util/limitrange/limitrange.go

```diff
@@ -138,18 +134,18 @@ func isSidecarContainer(container corev1.Container) bool {
 }
 
 // ValidatePodSpec verifies if the provided podSpec (ps) first into the boundaries of the summary (s).
-func (s Summary) ValidatePodSpec(ps *corev1.PodSpec, path *field.Path) []string {
-	reasons := []string{}
-	reasons = append(reasons, s.validatePodSpecContainers(ps.InitContainers, path.Child("initContainers"))...)
-	reasons = append(reasons, s.validatePodSpecContainers(ps.Containers, path.Child("containers"))...)
-	if containerRange, found := s[corev1.LimitTypePod]; found {
+func (s Summary) ValidatePodSpec(ps *corev1.PodSpec, path *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, s.validatePodSpecContainers(ps.InitContainers, path.Child("initContainers"))...)
+	allErrs = append(allErrs, s.validatePodSpecContainers(ps.Containers, path.Child("containers"))...)
+	if podRange, found := s[corev1.LimitTypePod]; found {
 		total := TotalRequests(ps)
```

Comment:
This conflicted with https://github.com/kubernetes-sigs/kueue/blob/06d4c763ce03891dc0f869e42d5557f9831f1682/pkg/util/limitrange/limitrange.go#L86